src/Native/LdaNative/data_block.cpp

```diff
@@ -45,8 +49,12 @@ namespace lda
 
     void LDADataBlock::Allocate(const int32_t num_document, const int64_t corpus_size)
     {
-        num_documents_ = num_document;
-        corpus_size_ = corpus_size;
+		num_documents_ = num_document;
+
+		if (corpus_size < 0 || static_cast<uint64_t>(corpus_size) > numeric_limits<size_t>::max())
+			bad_alloc();
+        corpus_size_ = static_cast<size_t>(corpus_size);
+
 
         offset_buffer_ = new int64_t[num_documents_ + 1]; // +1: one for the end of last document,
```

Comment:
Please check if `num_documents_ + 1` overflows.

Reply:
It should not:  num_documents_ is signed int 32, while size_t is either uint32 or uint64

---
In reply to: [276748897](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276748897) [](ancestors = 276748897)

Reply:
You are right. fixed - added check

---
In reply to: [276748897](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276748897) [](ancestors = 276748897)